### PR TITLE
Change licence expiration date pattern

### LINKED ISSUE #39136: Incorrect date formatting in License Expiration logs
**Describe the feature**:

**Elasticsearch version** (`bin/elasticsearch --version`):
8.0.0-SNAPSHOT
```
[root@73cce364f2b8 elasticsearch]# bin/elasticsearch --version
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
Version: 8.0.0-SNAPSHOT, Build: default/tar/dc02278/2019-02-19T08:51:47.611087Z, JVM: 11.0.2
```
**Plugins installed**: []

**JVM version** (`java -version`):

**OS version** (`uname -a` if on a Unix-like system):
Docker
```
[root@73cce364f2b8 elasticsearch]# uname -a
Linux 73cce364f2b8 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Description of the problem including expected versus actual behavior**:
Log message gives info on when my X-Pack license will expire, and the date string as formatted makes no sense. It reads: **License [will expire] on [Sunday, 3 10, 2019]**

`elasticsearch_1        | {"type": "server", "timestamp": "2019-02-19T16:27:25,723+0000", "level": "WARN", "component": "o.e.l.LicenseService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8", "cluster.uuid": "kNukTB2zQuW70QURs2g_dg", "node.id": "DjxjFDMjQ7WMbLrDsVxWmQ",  "message": "License [will expire] on [Sunday, 3 10, 2019].\n# If you have a new license, please update it. Otherwise, please reach out to\n# your support contact.\n# \n# Commercial plugins operate with reduced functionality on license expiration:\n# - security\n#  - Cluster health, cluster stats and indices stats operations are blocked\n#  - All data operations (read and write) continue to work\n# - watcher\n#  - PUT / GET watch APIs are disabled, DELETE watch API continues to work\n#  - Watches execute and write to the history\n#  - The actions of the watches don't execute\n# - monitoring\n#  - The agent will stop collecting cluster and indices metrics\n#  - The agent will stop automatically cleaning indices older than [xpack.monitoring.history.duration]\n# - graph\n#  - Graph explore APIs are disabled\n# - ml\n#  - Machine learning APIs are disabled\n# - logstash\n#  - Logstash will continue to poll centrally-managed pipelines\n# - beats\n#  - Beats will continue to poll centrally-managed configuration\n# - deprecation\n#  - Deprecation APIs are disabled\n# - upgrade\n#  - Upgrade API is disabled\n# - sql\n#  - SQL support is disabled\n# - rollup\n#  - Creating and Starting rollup jobs will no longer be allowed.\n#  - Stopping/Deleting existing jobs, RollupCaps API and RollupSearch continue to function."  }`

**Provide logs (if relevant)**:
```
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:23,860+0000", "level": "INFO", "component": "o.e.h.AbstractHttpServerTransport", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8",  "message": "publish_address {172.30.0.2:9200}, bound_addresses {0.0.0.0:9200}"  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:23,866+0000", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8",  "message": "started"  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:25,684+0000", "level": "INFO", "component": "o.e.l.LicenseService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8",  "message": "license [b458f5e9-f44b-4b36-be22-f9836445ddb8] mode [trial] - valid"  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:25,718+0000", "level": "INFO", "component": "o.e.g.GatewayService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8", "cluster.uuid": "kNukTB2zQuW70QURs2g_dg", "node.id": "DjxjFDMjQ7WMbLrDsVxWmQ",  "message": "recovered [16] indices into cluster_state"  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:25,723+0000", "level": "WARN", "component": "o.e.l.LicenseService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8", "cluster.uuid": "kNukTB2zQuW70QURs2g_dg", "node.id": "DjxjFDMjQ7WMbLrDsVxWmQ",  "message": "License [will expire] on [Sunday, 3 10, 2019].\n# If you have a new license, please update it. Otherwise, please reach out to\n# your support contact.\n# \n# Commercial plugins operate with reduced functionality on license expiration:\n# - security\n#  - Cluster health, cluster stats and indices stats operations are blocked\n#  - All data operations (read and write) continue to work\n# - watcher\n#  - PUT / GET watch APIs are disabled, DELETE watch API continues to work\n#  - Watches execute and write to the history\n#  - The actions of the watches don't execute\n# - monitoring\n#  - The agent will stop collecting cluster and indices metrics\n#  - The agent will stop automatically cleaning indices older than [xpack.monitoring.history.duration]\n# - graph\n#  - Graph explore APIs are disabled\n# - ml\n#  - Machine learning APIs are disabled\n# - logstash\n#  - Logstash will continue to poll centrally-managed pipelines\n# - beats\n#  - Beats will continue to poll centrally-managed configuration\n# - deprecation\n#  - Deprecation APIs are disabled\n# - upgrade\n#  - Upgrade API is disabled\n# - sql\n#  - SQL support is disabled\n# - rollup\n#  - Creating and Starting rollup jobs will no longer be allowed.\n#  - Stopping/Deleting existing jobs, RollupCaps API and RollupSearch continue to function."  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:26,794+0000", "level": "INFO", "component": "o.e.x.w.WatcherService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8", "cluster.uuid": "kNukTB2zQuW70QURs2g_dg", "node.id": "DjxjFDMjQ7WMbLrDsVxWmQ",  "message": "reloading watcher, reason [new local watcher shard allocation ids], cancelled [0] queued tasks"  }
[36melasticsearch_1        |[0m {"type": "server", "timestamp": "2019-02-19T16:27:27,889+0000", "level": "INFO", "component": "o.e.c.r.a.AllocationService", "cluster.name": "dev-docker-cluster", "node.name": "73cce364f2b8", "cluster.uuid": "kNukTB2zQuW70QURs2g_dg", "node.id": "DjxjFDMjQ7WMbLrDsVxWmQ",  "message": "Cluster health status changed from [RED] to [GREEN] (reason: [shards started [[babynames][1], [.kibana_1][0]] ...])."  }

```

### PR DESCRIPTION
**Pull request by pgomulka**
Due to migration from joda - java licence expiration 'full date' format
has to use 4-char pattern (MMMM). However since jdk9 the date with ROOT
locale will still return abbreviated days and month names.

closes #39136

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/license/LicenseServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        final String message = LicenseService.buildExpirationMessage(time, expired).toString();
+        if (expired) {
+            assertThat(message, startsWith("LICENSE [EXPIRED] ON [THU, NOV 15, 2018].\n"));
+        } else {
+            assertThat(message, startsWith("License [will expire] on [Thu, Nov 15, 2018].\n"));
+        }
+    }
+
+    public void testLogExpirationWarningOnJdk8() {
+        assumeTrue("this is for JDK9+", JavaVersion.current().equals(JavaVersion.parse("8")));
```
Comment (tvernum) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/license/LicenseServiceTests.java]:
```suggestion
        assumeTrue("this is for JDK8 only", JavaVersion.current().equals(JavaVersion.parse("8")));
```

**Review by tvernum — APPROVED** [review]
LGTM, but the `assumeTrue` message was copied incorrectly between tests.

**Comment by tvernum** [comment]
@pgomulka Sorry, I should have checked the labels when I reviewed this.
Can you add version labels and `> bug` please?

**Comment by pgomulka** [comment]
@tvernum good idea, that probably should be backported to 7.0 too right?

**Comment by tvernum** [comment]
Yes, `8.0.0`, `7.1.0` and `7.0.0`

**Comment by jpountz** [comment]
@pgomulka  I'm assuming there is nothing left to backport and removed the backport pending label.

**Pull request merged**